File: backend/app/schemas/publish.py
"""Publish asset schemas for API request/response."""

from datetime import datetime

from pydantic import BaseModel


class PublishAssetResponse(BaseModel):
    id: str
    platform: str
    cover_path: str | None = None
    cover_width: int | None = None
    cover_height: int | None = None
    cover_status: str = "pending"
    title: str
    description: str
    tags: str | None = None
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class PublishAssetUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    tags: str | None = None


class CoverRegenerateRequest(BaseModel):
    """Request to regenerate cover images with optional custom prompt."""
    prompt: str | None = None


class CoverPromptResponse(BaseModel):
    """Response containing the cover prompt."""
    prompt: str | None = None


class CoverPromptUpdate(BaseModel):
    """Request to update the cover prompt."""
    prompt: str

File: frontend/src/api/publish.ts
import apiClient from './client'

export interface PublishAsset {
  id: string
  platform: string
  cover_path: string | null
  cover_width: number | null
  cover_height: number | null
  cover_status: string
  title: string
  description: string
  tags: string | null
  status: string
  created_at: string
}

export interface PublishAssetUpdate {
  title?: string
  description?: string
  tags?: string
}

export interface CoverPrompt {
  prompt: string | null
}

export const publishApi = {
  getAssets(projectId: string) {
    return apiClient.get<PublishAsset[]>(`/projects/${projectId}/publish-assets`)
  },

  updateAsset(projectId: string, platform: string, data: PublishAssetUpdate) {
    return apiClient.put<PublishAsset>(`/projects/${projectId}/publish-assets/${platform}`, data)
  },

  useSegmentAsCover(projectId: string, segmentId: string) {
    return apiClient.post(`/projects/${projectId}/publish-assets/from-segment/${segmentId}`)
  },

  regenerateCover(projectId: string, prompt?: string) {
    return apiClient.post(`/projects/${projectId}/publish-assets/regenerate-cover`,
      prompt !== undefined ? { prompt } : null)
  },

  uploadCover(projectId: string, platform: string, file: File) {
    const formData = new FormData()
    formData.append('file', file)
    return apiClient.post<PublishAsset>(
      `/projects/${projectId}/publish-assets/${platform}/upload-cover`,
      formData,
      { headers: { 'Content-Type': 'multipart/form-data' } },
    )
  },

  getCoverPrompt(projectId: string) {
    return apiClient.get<CoverPrompt>(`/projects/${projectId}/publish-assets/cover-prompt`)
  },

  updateCoverPrompt(projectId: string, prompt: string) {
    return apiClient.put<CoverPrompt>(`/projects/${projectId}/publish-assets/cover-prompt`, { prompt })
  },
}
